File: 프로그래머스/3/258705. 산 모양 타일링/산 모양 타일링.js
function solution(n, tops) {
    
    var green = tops[0] == 0? 2 : 3;
    var purple = 1;
    
    for(let i=1; i<n; i++){
        var nextGreen = tops[i] == 0? 2*green + purple : 3*green + 2*purple;
        var nextPurple = green + purple;
        
        green = nextGreen%10007;
        purple = nextPurple%10007;
    }
    
    return (green+purple)%10007;
}

File: 프로그래머스/3/258705. 산 모양 타일링/산 모양 타일링.java
class Solution {
    public int solution(int n, int[] tops) {
        
        int green = tops[0] == 0 ? 2 : 3;
        int purple = 1;
        
        for(int i=1; i<n; i++){
            int nextGreen = tops[i] == 0 ? 2*green + purple : 3*green + 2*purple;
            int nextPurple = green + purple;
            
            green = nextGreen%10007;
            purple = nextPurple%10007;
        }
        
        return (green+purple)%10007;
    }
}